extensions/webdav/pkg/service/v0/service.go

```diff
@@ -67,10 +71,34 @@ func NewService(opts ...Option) (Service, error) {
 	}
 
 	m.Route(options.Config.HTTP.Root, func(r chi.Router) {
-		r.Get("/remote.php/dav/spaces/{id}/*", svc.SpacesThumbnail)
-		r.Get("/remote.php/dav/files/{id}/*", svc.Thumbnail)
-		r.Get("/remote.php/dav/public-files/{token}/*", svc.PublicThumbnail)
-		r.Head("/remote.php/dav/public-files/{token}/*", svc.PublicThumbnailHead)
+
+		r.Group(func(r chi.Router) {
+			r.Use(svc.DavUserContext())
+
+			r.Get("/remote.php/dav/spaces/{id}/*", svc.SpacesThumbnail)
+			r.Get("/dav/spaces/{id}/*", svc.SpacesThumbnail)
+
+			r.Get("/remote.php/dav/files/{id}/*", svc.Thumbnail)
+			r.Get("/dav/files/{id}/*", svc.Thumbnail)
+		})
+
+		r.Group(func(r chi.Router) {
+			r.Use(svc.DavPublicContext())
+
+			r.Head("/remote.php/dav/public-files/{token}/*", svc.PublicThumbnailHead)
+			r.Head("/dav/public-files/{token}/*", svc.PublicThumbnailHead)
+
+			r.Get("/remote.php/dav/public-files/{token}/*", svc.PublicThumbnail)
+			r.Get("/dav/public-files/{token}/*", svc.PublicThumbnail)
+		})
+
+		r.Group(func(r chi.Router) {
+			// currently not implemented
+			// If we implement this endpoint, we need to get the user from the access token.
+			// We decided against doing it, so that this service can be started without the JWT secret.
```

Comment:
hm why can't we just read the token and just pass it on when fetching the image ... or is the problem constructing a reference if all we have is the path? well ... for the /webdav endpoints we could fetch the users home path from the gateway and then combine it with the path in the request. we do have the token so ... that should work, shouldn't it?

Reply:
we could do a WhoAmI request to get the username